velox/functions/sparksql/String.h

```diff
@@ -1019,13 +1023,18 @@ struct ConvFunction {
     // When toBase is negative, converts to signed value. Otherwise, converts to
     // unsigned value. Overflow is allowed, consistent with Spark.
     if (toBase < 0) {
+      // This flag is used to make sure when we calculate the resultSize in
+      // `toChars` we always get a positive number. It is due to the
+      // `std::abs(min_int64)` would return a negative number.
+      auto isMinInt64Num =
+          unsignedValue == (uint64_t)std::numeric_limits<int64_t>::min();
       int64_t signedValue;
-      if (isNegativeInput) {
-        signedValue = -std::abs((int64_t)unsignedValue);
-      } else {
+      if (isMinInt64Num || !isNegativeInput) {
```

Comment:
It might be clearer to have a separate branch for isMinInt64Num with comments explaining what's going on.